### ISSUE TITLE
Overpowered MEM chips

### ISSUE THREAD

**Issue by Cwpute** [ochadenas/cpudefense#100]
So as a complete 180° turn from my previous issue with MEM chips #67, i now think these are way overpowered ! 🤡 espicially in the huge levels generated in Endless mode #96 .

While ACC have their price go up for every new ACC chip bought to compensate, and CLK becoming menacing for your CPU health as heat goes up, MEM chips are always interesting to place down.

They ignore attacker strenght, which instantly makes them a must-go for absurdly powered waves in later levels. And while they do recharge slowly, this can be improved by heavily upgrading their speed and range with recruits, and by lowering attackers rate and speed. And if it's still not enough, throw in a CLK chip and you're pretty much set for any situation.

* A quick fix would be to apply the same limitations the ACC chip gets, having the price for MEM go up for every new one bought (but that would only temporarily solve the issue in some early levels with MEM, as effective bit value goes down the more powerful attackers get).
* My prefered solution would **have MEM chips generate heat** for each attacker swallowed. This would probably require a bit of balancing of heat generation on CLK, but what i'd expect this solution to create is less use of MEM in early lane (so as to not excessively overheat), and more use as emergency or extraordinary chip placement so it still serves as a useful catch-all chip but also limits their overall number in a set level.

**Comment by Cwpute** [comment]
100th issue 🎉

**Comment by Cwpute** [comment]
I was thinking a good way to limit their effectiveness would be to require the player to tap on them once they have captured an attacker, before the chip can asorb a new attacker.

You would still be able to place down chips but would be required to manually intervene on each of them so they can attack again. So, a huge number of chips would require a huge amount of player interaction, limiting its power.
Contextually it works too ! as stored data doesn.t just go away, but need to be taken care of, so player interacting with the just could be interpreted as the user moving data around or away into antoher storage.

**Comment by ochadenas** [comment]
good idea. I'll try that.

**Comment by ochadenas** [comment]
I've implemented this suggestion. Seems to work quite well. For compensation I have reduced the cool down time (from 128 to 72). The whole mechanism still requires some fine-tuning, I suppose.

This is the new introductory text, I hope it is understandable:

Nous introduisons une nouvelle puce, la MEM (pour memory). Celle-ci prend un nombre et le stocke en toute sécurité dans la mémoire de l'ordinateur.

Le cadre plus large montre qu'un nombre se trouve à l'intérieur. Dans cet état, la puce MEM ne peut pas stocker un autre nombre. Elle doit d'abord être 'lue' (en la touchant).  Ce n'est qu'alors que commence son cycle de refroidissement et qu'elle peut absorber le prochain attaquant.

Les puces MEM sont puissantes, car elles peuvent neutraliser n'importe quel attaquant, mais elles demandent de l'attention et sont assez lentes.
